DDECal/DDECal.cc

```diff
@@ -306,11 +306,13 @@ void DDECal::initializeIDG(const ParameterSet& parset, const string& prefix) {
   std::string regionFilename = parset.getString(prefix + "idg.regions");
   std::vector<std::string> imageFilenames =
       parset.getStringVector(prefix + "idg.images");
-  itsFacetPredictor.reset(new FacetPredict(imageFilenames, regionFilename));
-  itsFacetPredictor->PredictCallback = std::bind(
-      &DDECal::idgCallback, this, std::placeholders::_1, std::placeholders::_2,
-      std::placeholders::_3, std::placeholders::_4);
-  itsDirections.resize(itsFacetPredictor->NDirections());
+  itsFacetPredictor = boost::make_unique<FacetPredict>(
+      imageFilenames, regionFilename,
+      [this](std::size_t row, std::size_t direction,
```

Comment:
This syntax is new to me, what happens here?

Reply:
I'm creating a lambda function that captures 'this'. When the facetpredictor calls it, the lambda forwards all arguments to idgCallback. It's somewhat equivalent to std::bind(idgCallback, this), but written using a more modern syntax.